### PR TITLE
don't add python and lua as default specs for private envs

### PR DESCRIPTION
**Pull request by kalefranz**

### CODE REVIEW

**Comment by mcg1969** [comment]
this makes a lot of sense

**Comment by kalefranz** [comment]
I think this is an innocuous change to include in 4.3, since we're really not using preferred_env all that heavily in production yet anyway.

File: conda/plan.py [hunk 1]
```diff
@@ -395,14 +395,16 @@ def is_root_prefix(prefix):
...
-def add_defaults_to_specs(r, linked, specs, update=False):
+def add_defaults_to_specs(r, linked, specs, update=False, prefix=None):
     # TODO: This should use the pinning mechanism. But don't change the API:
     # cas uses it.
     if r.explicit(specs):
         return
     log.debug('H0 specs=%r' % specs)
     names_linked = {r.package_name(d): d for d in linked if d in r.index}
     mspecs = list(map(MatchSpec, specs))
+    if is_private_env(prefix):
```
Comment (mcg1969) [inline conda/plan.py]: I know you've merged this, and technically it's fine, but you could have moved this up three lines.

Comment (kalefranz) [inline conda/plan.py]: Oh right. I'll fix it.

**Pull request merged**